Automattic/simperium
# auth: add dedicated reset_password handler

### Description

This PR adds a `reset_password` handler which ... actually already existed in the app? 😆  Not sure the value-added here as it's pretty much a subset of the existing `auth_http_update` logic. I suppose the only benefit is tweaking the endpoint to ensure that API usage is limited to _just_ password overrides?

Totally fine if this ends up being unnecessary, but was a nice little learning exercise nonetheless. 😄 

Context: Originally was looking at a `forgot_password` endpoint, but seems that isn't part of the [existing Simperium REST API](https://simperium.com/docs/reference/http/) and `reset_password` is what we wanted all along.

### To Test

1. Spin up simperium-local-dev with Docker and Erlang OTP with rebar3
2. If you don't have an existing app, user and developer to test with, then [create them](https://github.com/Automattic/simperium/blob/9e0b7b18a9a3a442ae8eccedfa5e3fabb4def46c/apps/simperium/src/db_models/sp_db_app.erl#L7-L19)
4. Identify the Admin API key and App ID for the app just created
5. Invoke a password reset with `curl -H 'Host: auth.simperium.com' -H 'X-Simperium-API-Key:<ADMIN-API-KEY>' http://localhost:8110/1/<APP-ID>/reset_password/ -d '{"username":"<USERNAME>","new_password":"<NEW_PASSWORD>"}' -i`
6. Verify the new password matches by using `sp_auth_password:passwords_match(...)` to validate the hashed and raw password.
